source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -282,7 +282,7 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
     auto const& host = result.host_info_;
     latchTime(decoder_callbacks_, DNS_END);
     if (!host.has_value() || !host.value()->address()) {
-      onDnsResolutionFail();
+      onDnsResolutionFail(host.has_value() ? *host : nullptr);
```

Comment:
How 'bout we just pass `host` down to `onDnsResolutionFail`? This will avoid making a copy of the shared pointer, I think.

Reply:
Copying shared pointer involve locking operations which is a bit of a bummer. Looking at the method, it looks like we don't actually use the value of the host at all, just the details. Maybe we could just pass in the details?

Reply:
sure